software/nosql/src/main/java/brooklyn/entity/nosql/couchbase/CouchbaseSyncGatewaySshDriver.java

```diff
@@ -92,10 +92,10 @@ public void launch() {
 
                 @Override
                 public boolean apply(@Nullable Entity entity) {
-                    if (entity instanceof CouchbaseNode && Boolean.TRUE.equals(entity.getAttribute(CouchbaseNode.IS_IN_CLUSTER))) {
-                        return true;
-                    }
-                    return false;
+                    // Must either be recognised by a cluster as added, or be the primary node in a cluster of ONE.
+                    return entity instanceof CouchbaseNode
+                      && (Boolean.TRUE.equals(entity.getAttribute(entity.IS_IN_CLUSTER))
+                      || Boolean.TRUE.equals(entity.getAttribute(entity.IS_PRIMARY_NODE)));
```

Comment:
Replace the 'entity' in `entity.IS_IN_CLUSTER` and `entity.IS_PRIMARY_NODE` with `CouchbaseNode`, otherwise this will not compile.